src/x86/x86_model.py

```diff
@@ -575,32 +581,33 @@ def speculate_instruction(emulator: Uc, address, size, model) -> None:
         # assemble source and destination operands of instruction
         # code duplication, with method speculate_fault(), could me moved in
         #    into separate method at some point
-        reg_src_operands = []
-        reg_dest_operands = []
+        reg_src_operands = set()
+        reg_dest_operands = set()
         # if the instruction accesses the memory, these registers are used in the address
-        address_regs = []
+        address_regs = set()
         for op in model.current_instruction.get_all_operands():
             if isinstance(op, RegisterOperand):
                 if op.src:
-                    reg_src_operands.append(X86TargetDesc.gpr_normalized[op.value])
+                    reg_src_operands.add(X86TargetDesc.gpr_normalized[op.value])
                 if op.dest:
-                    reg_dest_operands.append(X86TargetDesc.gpr_normalized[op.value])
+                    reg_dest_operands.add(X86TargetDesc.gpr_normalized[op.value])
             elif isinstance(op, MemoryOperand):
                 for sub_op in re.split(r'\+|-|\*| ', op.value):
                     if sub_op and sub_op in X86TargetDesc.gpr_normalized:
                         normalized = X86TargetDesc.gpr_normalized[sub_op]
-                        reg_src_operands.append(normalized)
-                        address_regs.append(normalized)
+                        reg_src_operands.add(normalized)
+                        address_regs.add(normalized)
             elif isinstance(op, FlagsOperand):
-                reg_src_operands.extend(op.get_read_flags())
-                reg_dest_operands.extend(op.get_write_flags())                
+                reg_src_operands.union(op.get_read_flags())
+                reg_dest_operands.union(op.get_write_flags())
         
-        # print('source operands:', reg_src_operands)  
-        # print('destination operands:', reg_dest_operands)        
-              
+        # print('source operands:', reg_src_operands)
+        # print('destination operands:', reg_dest_operands)
+        # print('Tainted:', model.reg_taints)
+
         # if source operands are not tainted, possible taint from destination operands 
         #   can be removed and control flow is returned
-        # print('intersection: ', reg_src_operands | model.reg_taints.keys())
+        # print('intersection: ', reg_src_operands & model.reg_taints.keys())
         if not (reg_src_operands & model.reg_taints.keys()):
```

Comment:
`reg_src_operands` and `reg_taints` are not instances of Set so we have built-ins for intersections and unions. You can't use | and & for set operations on List objects